pkg/operator/controllers/machine/machine_controller_test.go

```diff
@@ -0,0 +1,306 @@
+package machine
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	machinev1beta1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	maofake "github.com/openshift/machine-api-operator/pkg/generated/clientset/versioned/fake"
+	"github.com/operator-framework/operator-sdk/pkg/status"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	arofake "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned/fake"
+)
+
+func TestMachineReconciler(t *testing.T) {
+	newFakeMao := func(diskSize string, imagePublisher string, vmSize string) *maofake.Clientset {
+		master0 := &machinev1beta1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo-hx8z7-master-0",
+				Namespace: machineSetsNamespace,
+				Labels:    map[string]string{"machine.openshift.io/cluster-api-machine-role": "master"},
+			},
+			Spec: machinev1beta1.MachineSpec{
+				ProviderSpec: machinev1beta1.ProviderSpec{
+					Value: &runtime.RawExtension{
+						Raw: []byte(`{
+"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
+"kind": "AzureMachineProviderSpec",
+"osDisk": {
+"diskSizeGB": 512
+},
+"image": {
+"publisher": "azureopenshift",
+"offer": "aro4"
+},
+"vmSize": "Standard_D8s_v3"
+}`)},
+				},
+			},
+		}
+
+		master1 := &machinev1beta1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo1-hx8z7-master-1",
+				Namespace: machineSetsNamespace,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Kind: "MachineSet",
+					},
+				},
```

Comment:
Same as here: https://github.com/Azure/ARO-RP/pull/1505#discussion_r658669482. You said that these were leftovers. There are more in this file: if they are not important in the test - we should get rid of them.